src/framework/mpas_stream_manager.F

```diff
@@ -4148,8 +4153,18 @@ end subroutine mpas_build_stream_filename !}}}
     !-----------------------------------------------------------------------
     subroutine build_stream(stream, direction, allFields, allPackages, timeLevelIn, mgLevelIn, ierr) !{{{
 
+        use iso_c_binding, only : c_int, c_char
+
         implicit none
 
+        interface
+          subroutine gen_random(len, id) bind(c)
+             use iso_c_binding, only : c_int, c_char
+             integer (c_int), intent(in), value :: len
+             character (c_char), intent(inout) :: id
```

Comment:
I think `id` should be a character array, as we did for `fname` in the `read_geogrid` interface.

Reply:
Sounds good to me. Updated the deceleration here as updated the Fortran deceleration interface within `random_id.c`.